frontend/src/services/protect-csv.service.ts

```diff
@@ -36,6 +36,49 @@ async function transformRows(
   return transformed.join('\n')
 }
 
+async function chunkSize(file: File, template: string): Promise<number> {
+  const defaultChunkSize = 10000000 // 10 Mb
+  const minChunkSize = 5000000 // 5Mb minimum needed for uploading each part
+  let numChars = template.length
+  let rowChars = 0
+  return new Promise((resolve) => {
+    Papa.parse(file, {
+      header: true,
+      delimiter: ',',
+      step: function (_, parser: Papa.Parser) {
+        // Checks first row only
+        parser.pause()
+        parser.abort()
+      },
+      complete: function (results) {
+        // We can also check for recipient and password here.
+        // results.data will contain 1 row of results because we aborted on the first step
+        const row = Object.entries(results.data) as Array<[string, string]>
+        for (const [field, value] of row) {
+          if (/^[a-zA-Z0-9\s-_'"/]+$/.test(field)) {
+            // field is of an acceptable format
+            const regexp = new RegExp(`\\s+${field}\\s+`, 'gi')
+            const numReplacements = template.match(regexp)?.length || 0
```

Comment:
Should we use the templating module instead of regexp? For example, the template could be `hi {{recipient}}, your secret code is {{ password }}` and the regex will only catch `password` and not `recipient`.